File: src/SamuraiServer/SamuraiServer.Data/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SamuraiServer.Data
{
    public class Player
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string ApiKey { get; set; }

        public int Wins { get; set; }
        public int GamesPlayed { get; set; }

        public DateTime LastSeen { get; set; }

        public bool IsOnline { get; set; }

        public bool IsActive { get; set; }
    }
}

File: src/Samurai.Client.JS/models/player.js
define(function () {
  function Player(opt) {
    this.initialize(opt);
  }

  Player.prototype.initialize = function (opt) {
    var key = null,
    default_args = {
      'Id' : null,
      'Name' : '',
      'ApiKey' : '',
      'Wins' : 0,
      'GamesPlayed' : 0,
      'LastSeen' : new Date(),
      'IsOnline' : false,
      'IsActive' : false
    };

    opt || (opt = default_args);

    for(key in default_args) {
      if(typeof opt[key] == "undefined")
        opt[key] = default_args[key];
    }

    // opt[] has all the data - user provided and optional.
    for(key in opt) {
      this[key] = opt[key];
    }
  };

  return Player;
});